app/controllers/katello/api/v2/products_controller.rb

```diff
@@ -47,7 +47,15 @@ def index
 
     def index_relation
       query = Product.readable.where(:organization_id => @organization.id)
-      query = query.where(:provider_id => @organization.anonymous_provider.id) if params[:custom]
+
+      if params[:custom]
+        if Foreman::Cast.to_bool(params[:custom])
+          query = query.where(:provider_id => @organization.anonymous_provider.id)
+        else
+          query = query.where.not(:provider_id => @organization.anonymous_provider.id)
+        end
+      end
```

Comment:
I think this makes the parameter somewhat confusing (as well as changing the meaning of the api) as it now has 3 states:
1.  set to true
2.  set to false
3.  not set

we typically don't this (the filters are usually on or off), but i'm okay with it as long as we document that it has 3 state in the api params

Reply:
sure thing, all fair points, I can add the documentation.

The only other alternative I can see is have two params
`custom_only` to just return custom products
`redhat_only` to only return redhat products
(these are the first names that came to mind so take them with a grain of salt)

but if we can support accepting booleans, I don't see why we couldn't accept the 3 states

Reply:
yeah, these filters are usually on or off, we've just not accepted an 'undefined' state.  In cases where we have multiple modes, we have supported something like 'mode=foo' where we document what options you can pass to the 'mode' param  (mode is just an example here).  I think i prefer the 'redhat_only' option myself, but don't have super strong opinions as long as its well documented and doesn't change the current api